web/docs/data-model/operations/actions.md

```diff
@@ -182,27 +186,62 @@ export const markTaskAsDone: MarkTaskAsDone<Pick<Task, 'id'>, void> = (
 }
 ```
 
+<SuperjsonNote />
+
+#### Type support for Actions
+
 Wasp automatically generates the types `CreateTask` and `MarkTaskAsDone` based on the declarations in your Wasp file:
 
 - `CreateTask` is a generic type that Wasp automatically generated based on the Action declaration for `createTask`.
 - `MarkTaskAsDone` is a generic type that Wasp automatically generated based on the Action declaration for `markTaskAsDone`.
 
-You can use these types to specify the Action's input and output types.
+Use these types to type the Action's implementation.
+Annotating Actions is optional but very helpful.
+Doing so properly types the Action's context.
 
-The Action `createTask` expects to get an object of type `{ description: string }` and returns the newly created task (an object of type `Task`).
+In this case, TypeScript will know the `context.entities` object must include the `Task` entity.
+TypeScript also knows whether the `context` object includes user information (it depends on whether your Action uses auth).
 
-The Action `markTaskAsDone`, expects an object of type `{ id: number }` and doesn't return anything (i.e., its return type is `void`).
+The generated types are generic and accept two optional type arguments: `Input` and `Output`.
 
-We've derived most of the payload types from the type `Task`.
+1. `Input` - The argument (i.e., payload) received by the Action function.
+2. `Output` - The Action function's return type.
 
-Annotating the Actions is optional, but highly recommended. Doing so enables **full-stack type safety**. We'll see what this means when calling the Action from the client.
+Use these type arguments to type the Action's inputs and outputs.
 
-:::tip
-Wasp uses [superjson](https://github.com/blitz-js/superjson) under the hood. In other words, you don't need to limit yourself to only sending and receiving JSON payloads.
+For example, the above code says that the Action `createTask` expects an object with the new task's description (its input type is `Pick<Task, 'description'>`) and returns the new task (its output type is `Task`).
+On the other hand, the Action `markTaskAsDone` expects an object of type `Pick<Task, 'id'>`. This type is derived from the `Task` entity type.
+
+Suppose you don't care about typing the Action's inputs and outputs. In that case, you can omit both type arguments, and TypeScript will infer the most general types (i.e., `never` for the input, `unknown` for the output.).
 
-Send and receive any superjson-compatible payload (e.g., Dates, Sets, Lists, circular references, etc.) and let Wasp take care of the (de)serialization.
+Even though specifying `Input` or `Output` is completely optional, we highly recommend you specify them. Doing so gives you:
```

Comment:
When I see this amount of text being added - I wonder if some parts of this could be behind `<details>` blocks to keep the cognitive load lower. I'm not sure, but just a thing to think about.